### PR TITLE
[UK-1013] Fix file view error and add debouncing

### PR DESCRIPTION
**Pull request by HoonBaek**
## For Internal Contributors

[UK-1013](https://sendbird.atlassian.net/browse/UK-1013)

## Description Of Changes

* Add debouncing logic to the MessageSearch component
* Add optional chaining for isResendable of message
* Add optional chaining for `emojiCategories` of message

## Types Of Changes

What types of changes does your code introduce to this project?
Put an `x` in the boxes that apply_

- [x] Bugfix
- [x] New feature
- [ ] Documentation (correction or otherwise)
- [ ] Cosmetics (whitespace, appearance (ex) Prettier)
- [ ] Build configuration
- [ ] Improvement (refactor code)

### CODE REVIEW

File: src/smart-components/MessageSearch/hooks/useSearchStringEffect.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { useEffect } from 'react';
+
+interface DynamicParams {
+  searchString: string;
+}
+interface StaticParams {
+  setRequestString(string: string): void;
+}
+
+const DEBOUNCING_TIME = 500;
```
Comment (HoonBaek) [inline src/smart-components/MessageSearch/hooks/useSearchStringEffect.ts:10]: @sravan-s Do you think it's good to get a debouncing time from the props of MessageSearch component?

Comment (sravan-s) [reply src/smart-components/MessageSearch/hooks/useSearchStringEffect.ts:10]: I can add that prop during modularization

File: src/ui/MessageItemMenu/index.tsx [hunk 2]
```diff
@@ -44,8 +44,8 @@ export default function MessageItemMenu({
   const showMenuItemCopy: boolean = isUserMessage(message as UserMessage);
   const showMenuItemReply: boolean = false && !isFailedMessage(channel, message) && !isPendingMessage(channel, message);
   const showMenuItemEdit: boolean = (isUserMessage(message as UserMessage) && isSentMessage(channel, message) && isByMe);
-  const showMenuItemResend: boolean = (isFailedMessage(channel, message) && message.isResendable() && isByMe);
-  const showMenuItemDelete: boolean = (isSentMessage(channel, message) && isByMe);
+  const showMenuItemResend: boolean = (isFailedMessage(channel, message) && message?.isResendable?.() && isByMe);
+  const showMenuItemDelete: boolean = isByMe;
```
Comment (sravan-s) [inline src/ui/MessageItemMenu/index.tsx:48]: isSentMessage(channel, message) -> No errors while deleting unsent message?

File: src/smart-components/MessageSearch/index.tsx [hunk 3]
```diff
@@ -112,8 +110,10 @@ function MessageSearch(props: Props): JSX.Element {
     { sdk, logger, messageSearchDispathcer },
   );
 
+  useSearchStringEffect({ searchString }, { setRequestString });
```
Comment (sravan-s) [inline src/smart-components/MessageSearch/index.tsx:113]: Designing something like 
`const requestString = useSearchStringEffect({ searchString }); `

would have been better

**Review by sravan-s — APPROVED** [review]

File: src/smart-components/MessageSearch/index.tsx (hunk 3 above)
Comment (HoonBaek) [reply src/smart-components/MessageSearch/index.tsx:113]: Good suggestion!

File: src/ui/MessageItemMenu/index.tsx (hunk 2 above)
Comment (HoonBaek) [reply src/ui/MessageItemMenu/index.tsx:48]: There was no new appeared error when I remove this condition

**Review by sravan-s — APPROVED** [review]

**Pull request merged**